core/src/main/java/org/springframework/security/authentication/AbstractAuthenticationToken.java

```diff
@@ -112,8 +112,8 @@ public void eraseCredentials() {
 	}
 
 	private void eraseSecret(Object secret) {
-		if (secret instanceof CredentialsContainer) {
-			((CredentialsContainer) secret).eraseCredentials();
+		if (secret instanceof CredentialsContainer secret) {
```

Comment:
The variable name `secret` is already defined in the scope, can you please rename it to maybe `container`?